source/robot_model/test/tests/test_kinematics.cpp

```diff
@@ -0,0 +1,152 @@
+                                                                                              0.422247,
+                                                                                              0.800825,
+                                                                                              -0.424709)));
+    twist << -0.906095, -0.344373, -0.092142, 0.322764, -0.096389, 0.429407;
+    test_velocity_fk_expects.emplace_back(state_representation::CartesianTwist("ee", twist));
+  }
+};
+
+TEST_F(RobotModelKinematicsTest, TestForwardGeometryJointStateSize) {
+  state_representation::JointState dummy = state_representation::JointState(robot_name, 6);
+  EXPECT_THROW(franka->forward_geometry(dummy), exceptions::InvalidJointStateSizeException);
+}
+
+TEST_F(RobotModelKinematicsTest, TestForwardGeometryEE) {
+  EXPECT_EQ(franka->forward_geometry(joint_state).get_position(),
+            franka->forward_geometry(joint_state, "panda_link8").get_position());
+}
+
+TEST_F(RobotModelKinematicsTest, TestForwardGeometryInvalidFrameName) {
+  EXPECT_THROW(franka->forward_geometry(joint_state, "panda_link99"), exceptions::FrameNotFoundException);
+}
+
+TEST_F(RobotModelKinematicsTest, TestForwardGeometry) {
+  for (std::size_t config = 0; config < test_configs.size(); ++config) {
+    state_representation::CartesianPose ee_pose = franka->forward_geometry(test_configs[config]);
+    EXPECT_TRUE(ee_pose.get_position().isApprox(test_fk_ee_expects.at(config).get_position()));
+    EXPECT_TRUE(ee_pose.get_orientation().isApprox(test_fk_ee_expects.at(config).get_orientation()));
+    state_representation::CartesianPose link4_pose = franka->forward_geometry(test_configs[config], "panda_link4");
+    EXPECT_TRUE(link4_pose.get_position().isApprox(test_fk_link4_expects.at(config).get_position()));
+    EXPECT_TRUE(link4_pose.get_orientation().isApprox(test_fk_link4_expects.at(config).get_orientation()));
+  }
+}
+
+TEST_F(RobotModelKinematicsTest, TestForwardKinematics) {
+  for (std::size_t config = 0; config < test_configs.size(); ++config) {
+    state_representation::CartesianTwist ee_twist = franka->forward_kinematic(test_configs[config]);
+    EXPECT_TRUE(ee_twist.get_linear_velocity().isApprox(test_velocity_fk_expects.at(config).get_linear_velocity()));
+    EXPECT_TRUE(ee_twist.get_angular_velocity().isApprox(test_velocity_fk_expects.at(config).get_angular_velocity()));
+  }
+}
+
+TEST_F(RobotModelKinematicsTest, TestInverseKinematics) {
+  for (std::size_t config = 0; config < test_configs.size(); ++config) {
+    state_representation::CartesianTwist ee_twist = franka->forward_kinematic(test_configs[config]);
+    state_representation::JointVelocities joint_twist = franka->inverse_kinematic(test_configs[config], ee_twist);
+    for (std::size_t joint = 0; joint < franka->get_number_of_joints(); ++joint) {
+      EXPECT_NEAR(joint_twist.get_velocities()[joint], test_configs[config].get_velocities()[joint], tol);
```

Comment:
You can also use  `EXPECT_TRUE(joint_twist.get_velocities().isApprox(test_configs[config].get_velocities()))`. At the end it is a matter whether or not we want additional details on which joint fails or simply knowing that the full IK fails. I don't think knowing specifically which joint is having an issue changes anything though.